src/ui/public/pattern_checker/pattern_checker.js

```diff
@@ -0,0 +1,56 @@
+import uiModules from 'ui/modules';
+import template from './pattern_checker.html';
+import './pattern_checker.less';
+
+const module = uiModules.get('kibana');
+
+module.directive('patternChecker', function () {
+  return {
+    restrict: 'E',
+    template: template,
+    controllerAs: 'checker',
+    bindToController: true,
+    scope: {
+      pattern: '=',
+      buttonLabel: '='
+    },
+    controller: function (es, Notifier, $scope, $timeout) {
+      let validationTimeout;
+      this.isValidating = false;
+
+      var notify = new Notifier({
+        location: 'Add Data'
+      });
+
+      this.toggleValidation = () => {
+        if (!this.isValidating) {
+          this.isValidating = true;
+          this.validateInstall();
+        }
+        else {
+          $timeout.cancel(validationTimeout);
+          this.isValidating = false;
+          this.validationResults = '';
+        }
+      };
+
+      this.validateInstall = () => {
+        es.count({
```

Comment:
@rashidkpc I asked myself the same question, I didn't know if it made sense for something so simple but if we really want to eliminate the ES client completely then I guess it does.

Any opinions on what the endpoint for such an API should be? I could see any of the following being viable options:

`localhost:5601/api/kibana/{indexPatternId}/_count`
(mirrors ES syntax)

`localhost:5601/api/kibana/ingest/{indexPatternId}/_count`
(with the ingest API we've sort of established the ingest config as a higher level concept than an index pattern, so maybe it belongs under ingest?)

`localhost:5601/api/kibana/index_patterns/{indexPatternId}/_count`
(this one feels weird but I include it because it's a way to namespace the API without re-using ingest)

Something else?